pkg/devfile/adapters/common/command.go

```diff
@@ -84,6 +88,33 @@ func getCommand(data data.DevfileData, commandName string, groupType common.Devf
 	return
 }
 
+// validateCommandsForGroup validates the commands in a devfile for a group
+// 1. multiple commands belonging to a single group should have at least one default
+// 2. multiple commands belonging to a single group cannot have more than one default
+func validateCommandsForGroup(data data.DevfileData, groupType common.DevfileCommandGroupType) error {
+
+	commands := getCommandsForGroup(data, groupType)
+
+	defaultCommandCount := 0
+
+	if len(commands) > 1 {
+		for _, command := range commands {
+			if command.Exec.Group.IsDefault {
+				defaultCommandCount++
+			}
+		}
+	} else {
+		// if there is only one command, it is the default command for the group
+		defaultCommandCount = 1
+	}
+
+	if defaultCommandCount != 1 {
+		return fmt.Errorf("there should be one default command for command group %v", groupType)
+	}
```

Comment:
`there should be only one default command for command group`

`there should be one default command for command group` signifying that no default command is present.

Reply:
updated

Reply:
actually this condition checks for both missing and multiple defaults

Reply:
IMO, we could return different error for  `defaultCommandCount=0` & `defaultCommandCount >1`
or 
`there should be exactly one default command for command group `

Reply:
Yeah makes sense, updated the err statements based on the default command count